api/clroni/clroni/Frame.cs

```diff
@@ -62,10 +63,17 @@ protected Frame()
         //    // Copy the memory
         //    Buffer.MemoryCopy(data.ToPointer(), frame->data, num_bytes, data_size);
         //}
+        private long mempress = 0;
+        public void addMemoryPressure(long mem)
+        {
+            mempress += mem;
+            GC.AddMemoryPressure(mem);
+        }
 
         protected override bool ReleaseHandle()
         {
             lib.NativeMethods.oni_destroy_frame(handle);
+            GC.RemoveMemoryPressure(mempress);
```

Comment:
If memory pressure is added in the constructor, we could just use the size of the frame instead of needing an adittional variable.